hcmanager/utils_test.go

```diff
@@ -30,43 +30,97 @@ import (
 
 var _ = Describe("HCManager", func() {
 
+	type testCaseFetchBMH struct {
+		namespace     string
+		hosts         []runtime.Object
+		expectedError bool
+	}
+
 	DescribeTable("Test fetch BaremetalHost list",
-		func(namespace string) {
-			hostTest := getHosts()
-			c := fakeclient.NewFakeClientWithScheme(setupSchemeMm(), hostTest...)
+		func(tc testCaseFetchBMH) {
+			c := fakeclient.NewFakeClientWithScheme(setupSchemeMm(), tc.hosts...)
 			hcManager := NewHardwareClassificationManager(c, klogr.New())
-			result, _, err := hcManager.FetchBmhHostList(namespace)
-			if err != nil {
+			result, _, err := hcManager.FetchBmhHostList(tc.namespace)
+			if tc.expectedError {
 				Expect(len(result)).To(BeZero())
```

Comment:
you should expect err to have occurred here.

Reply:
Please add an expect here for err to have occurred.

Reply:
done. @maelk @digambar15 kindly review